src/components/ListItem.jsx

```diff
@@ -1,14 +1,43 @@
 import { updateItem } from '../api';
 import { subtractDates } from '../utils';
 import { Timestamp } from 'firebase/firestore';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 import './ListItem.css';
 
-export function ListItem({ name, listPath, id, dateLastPurchased }) {
+export function ListItem({
+	name,
+	listPath,
+	id,
+	dateLastPurchased,
+	dateNextPurchased,
+	totalPurchases,
+	dateCreated,
+}) {
 	const todaysDate = Timestamp.now();
 
+	// if dateLastPurchased is true subtract it from dateNextPurchased, else subtract dateCreated from dateNextPurchased to get the estimated number of days till next purchase
+	const previousEstimate = Math.ceil(
+		(dateNextPurchased.toDate() -
```

Comment:
When I navigated to the Groceries list I got this error:
![Screenshot 2024-03-09 at 7 40 05 AM](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/25274c81-c40e-4aa1-b057-816577c0d090)
There were several items on the list that were created before the logic for all of the date fields (dateCreated, dateLastPurchased, etc) was implemented, so the `previousEstimate` function was getting an undefined variable.
I do not think that this will be an issue now that the logic exists for all of the date fields. I am not sure, but it might be a good idea to add some error handling in case somehow one of these fields is not calculated? What do you think?